mirgecom/inviscid.py

```diff
@@ -128,32 +128,183 @@ def inviscid_facial_flux(discr, state_tpair, local=False):
         the face normals as required by the divergence operator for which they
         are being computed.
     """
-    actx = state_tpair.int.array_context
-    dd = state_tpair.dd
-    dd_all_faces = dd.with_dtag("all_faces")
+    actx = state_pair.int.array_context
+    lam = actx.np.maximum(state_pair.int.wavespeed, state_pair.ext.wavespeed)
+    from mirgecom.flux import num_flux_lfr
+    return num_flux_lfr(f_minus_normal=inviscid_flux(state_pair.int)@normal,
+                        f_plus_normal=inviscid_flux(state_pair.ext)@normal,
+                        q_minus=state_pair.int.cv,
+                        q_plus=state_pair.ext.cv, lam=lam)
 
-    flux_tpair = TracePair(dd,
-                           interior=inviscid_flux(state_tpair.int),
-                           exterior=inviscid_flux(state_tpair.ext))
 
-    # This calculates the local maximum eigenvalue of the flux Jacobian
-    # for a single component gas, i.e. the element-local max wavespeed |v| + c.
-    w_int = state_tpair.int.speed_of_sound + state_tpair.int.speed
-    w_ext = state_tpair.ext.speed_of_sound + state_tpair.ext.speed
-    lam = actx.np.maximum(w_int, w_ext)
+def inviscid_flux_hll(state_pair, gas_model, normal, **kwargs):
```

Comment:
Should this and `inviscid_flux_rusanov` maybe be called `inviscid_facial_flux_<rusanov/hll>`, since they return the normal flux?

Reply:
I am not opposed to such a renaming, but fwiw, the fluxes we are using according to their definitions in the approximate Riemann solver references (e.g., https://edanya.uma.es/NSPDE/cursos/toro/2-HLL+HLLC.pdf), are defined as interface-normal fluxes already.

Reply:
That's fine, I was just thinking we should try to stick to a consistent naming convention in the code if we can, i.e., `*_flux` for vectors, `*_facial_flux` for normal fluxes.

Reply:
([72bc8a5](https://github.com/illinois-ceesd/mirgecom/pull/648/commits/72bc8a5f9210852630243c1d1d8eb03a1d9421d6))